webapp/src/snippetBuilder.tsx

```diff
@@ -105,19 +158,18 @@ export class SnippetBuilder extends data.Component<SnippetBuilderProps, SnippetB
 
     /**
      * 
-     * @param output - Accepts an output to convert to markdown
      * This attaches three backticks to the front followed by an output type (blocks, lang)
      * The current output is then tokenized and three backticks are appended to the end of the string.
      */
-    generateOutputMarkdown(tsOutput: string) {
-        const { config } = this.state;
+    generateOutputMarkdown = pxt.Util.debounce(() => {
+        const { config, tsOutput } = this.state;
         // Attaches starting and ending line based on output type
         let md = `\`\`\`${config.outputType}\n`;
         md += this.replaceTokens(tsOutput);
         md += `\n\`\`\``;
 
-        return md
-    }
+        this.setState({ mdOutput: md });
+    }, 300, false);
```

Comment:
Test 300 on slow machine, you might want to increase this in “light” mode

Reply:
Good idea, will do. I'm going to be focusing a lot on how it works on different machines/platforms going forward.